script.js

```diff
@@ -1,4 +1,184 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+//BASIC BEAT THAT
+// 2 players
+// Player rolls 2 dice and see's 2 options.
+// Player selects 1 out of 2 options for 1st & 2nd order
+// Game registers Player 1's selection & calls of player 2 to roll & select
+// Game then shows winner, some stats, triggers next round, etc
+
+//[HF] Dice Roll
+var diceRoll = function () {
+  var randomNumber = Math.random() * 6;
+  var randomInteger = Math.floor(randomNumber);
+  var integerNotZero = randomInteger + 1;
+  return integerNotZero;
 };
+//[HF] Result comparison
+var finalResult = function () {
+  var playerChoices = `${player1} you chose ${
+    p1choice[p1choice.length - 1]
+  } <br> ${player2} you chose ${p2choice[p2choice.length - 1]}`;
+  // Both Draw
+  if (p1choice[p1choice.length - 1] == p2choice[p2choice.length - 1]) {
+    displayMsg = `ITS A DRAW! <br> ${playerChoices}`;
+  }
+  // P1 Wins
+  if (p1choice[p1choice.length - 1] > p2choice[p2choice.length - 1]) {
+    displayMsg = `${player1} WINS! <br> ${playerChoices} <br> click submit to play again!`;
+  }
+  // P2 Wins
+  if (p1choice[p1choice.length - 1] < p2choice[p2choice.length - 1]) {
+    displayMsg = `${player2} WINS! <br> ${playerChoices} <br> click submit to play again!`;
+  }
+  return displayMsg;
+};
+
+//[HF]
+
+//Global Variables
+var player1 = "";
+var player2 = "";
+var gameMode = "P1 name";
+var p1options = [];
+var p2options = [];
+var p1choice = [];
+var p2choice = [];
+
+function main(input) {
+  // When mode : waiting for player 1 name,  player 1 does not enter name
+  if (gameMode == "P1 name" && input == "") {
+    return "Player 1 please enter your name leh.";
+  }
+  // When player 1 enters name. Mode changes to : Prompt player 2 to enter name
+  if (input == input && gameMode == "P1 name") {
+    gameMode = "P2 name";
+    console.log("should be P2");
+    console.log(gameMode);
+    player1 = input;
+    return `Welcome ${player1}. Player 2 please enter your name.`;
+  }
+  // When player 2 enters a blank/or clicks submit by accident
+  if (gameMode == "P2 name" && input == "") {
```

Comment:
you can simplify your code by creating a parent if condition for "if gameMode == "p2 name" and nest the different conditionals inside, so u dont have to keep repeating "if gameMode == "p2 name"